src/redis_index.c

```diff
@@ -357,7 +357,7 @@ IndexReader *Redis_OpenReader(RedisSearchCtx *ctx, RSQueryTerm *term, DocTable *
     }
   }
 
-  if (!idx->numDocs) {
+  if (!idx->numDocs || (!(idx->fieldMask & fieldMask) && (ctx->spec->flags & Index_StoreFieldFlags))) {
     // empty index! pass
```

Comment:
Maybe just fix this comment, there are more cases that we skip now. Maybe we can extract it to a function? `ShouldProcessesIdx(...)`?